src/cmd/policy.go

```diff
@@ -136,6 +144,61 @@ func RegoFuncGetGithubRepo(ctx rego.BuiltinContext, a, b *ast.Term) (*ast.Term,
 
 	reader := strings.NewReader(response.String())
 	v, err := ast.ValueFromReader(reader)
+
+	return ast.NewTerm(v), nil
+}
+
+func RegoFuncGetGitlabRepo(ctx rego.BuiltinContext, a *ast.Term) (*ast.Term, error) {
+
+	var path string
+	if err := ast.As(a.Value, &path); err != nil {
+		return nil, err
+	}
+
+	if path == "" {
+		log.Error().Msgf("opslevel.repo.gitlab(\"%s\") failed: Please provide a valid Gitlab project path", path)
+		return nil, nil
+	}
+
+	escapedPath := url.QueryEscape(path)
+	gitlabToken := viper.GetString("gitlab-token")
+	gitlabAPIUrl := fmt.Sprintf("https://gitlab.com/api/v4/projects/%s", escapedPath)
+
+	type GitlabRepoMetadata struct {
```

Comment:
Went with a struct this time since I had to make 2 calls this time and thought it might be better to explicitly build up an object in this case? Not sure if that's the right approach or if I should just add languages result to the first response.

Reply:
Ok - this is a good test to see which route works better long term.

should the struct be defined outside the function?
should the struct be private (first letter lowercase)?
How do we intend to document the available fields?

Reply:
Yeah, this should be defined outside of the function and should be private for now unless we need to end up using it elsewhere.  I imagine it'll have to be documented when we document each function that we've added to Rego?

Reply:
We don't have do this for this ticket but we might want to enqueue a backlog ticket for documenting our custom functions.